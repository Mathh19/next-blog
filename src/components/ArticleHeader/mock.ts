import { ArticleHeaderProps } from '.';

export default {
  title:
    'Sint deserunt officia sit ipsum. Minim commodo sit ea labore sit sit ex ad tempor esse labore.',
  excerpt:
    'Proident sunt est enim fugiat sit incididunt sunt veniam adipisicing amet aliquip.',
  cover:
    'https://res.cloudinary.com/dlizakp2a/image/upload/v1614696630/Frozen_sunset_on_the_lake_by_Manuel_Arslanyan_9f9cd8ea10.jpg',
  createdAt: '2022-09-02T15:16:09.246+00:00',
  author: {
    id: '123456789',
    attributes: {
      slug: 'matheus-freitas',
      name: 'Matheus Freitas',
    },
  },
  categories: [
    {
      id: 'asdslgyh65',
      attributes: {
        name: 'React',
        slug: 'react',
      },
    },
    {
      id: 'asdslgyh66',
      attributes: {
        name: 'JavaScript',
        slug: 'javascript',
      },
    },
  ],
} as ArticleHeaderProps;
