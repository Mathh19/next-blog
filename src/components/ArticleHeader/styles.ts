import styled, { css } from 'styled-components';
import { Title as HeadingStyles } from '../Heading/styles';

export const Wrapper = styled.header`
  ${({ theme }) => css`
    padding-bottom: ${theme.spacings.xlarge};
    margin-bottom: ${theme.spacings.xlarge};
    border-bottom: 0.1rem solid ${theme.colors.mediumGray};

    ${HeadingStyles} {
      margin: 0;
      margin-bottom: ${theme.spacings.medium};
    }
  `}
`;

export const Excerpt = styled.p`
  ${({ theme }) => css`
    margin: ${theme.spacings.medium} 0;
    font-size: ${theme.font.sizes.medium};
  `}
`;

export const CoverContainer = styled.div`
  ${({ theme }) => css`
    display: flex;
    justify-content: center;
    margin-bottom: ${theme.spacings.medium};
  `}
`;

export const Cover = styled.img`
  ${css`
    max-width: 100%;
    display: block;
  `}
`;
