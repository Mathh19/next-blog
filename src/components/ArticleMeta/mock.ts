import { ArticleMetaProps } from '.';

export default {
  createdAt: '2022-09-02T15:16:09.246+00:00',
  author: {
    id: '1234567890',
    attributes: {
      slug: 'matheus-freitas',
      name: 'Matheus Freitas',
    },
  },
  categories: [
    {
      id: 'asdslgyh65',
      attributes: {
        name: 'React',
        slug: 'react',
      },
    },
    {
      id: 'asdslgyh66',
      attributes: {
        name: 'JavaScript',
        slug: 'javascript',
      },
    },
  ],
} as ArticleMetaProps;
