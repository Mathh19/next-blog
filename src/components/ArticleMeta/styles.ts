import styled, { css } from 'styled-components';

export const Wrapper = styled.article`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.small};
    color: ${theme.colors.darkGray};
    font-style: italic;

    .categories span::after {
      content: ', ';
    }
    .categories span:last-child::after {
      content: '';
    }

    a {
      color: ${theme.colors.secondary};
      text-decoration: none;
      transition: all 300ms ease-in-out;

      &:hover {
        filter: brightness(50%);
      }
    }
  `}
`;
