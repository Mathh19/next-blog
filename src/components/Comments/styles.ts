import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${({ theme }) => css`
    max-width: ${theme.sizes.content};
    width: 100%;
    margin: ${theme.spacings.large} auto;
    padding: 0 ${theme.spacings.large};
  `}
`;

export const NoComments = styled.div`
  ${({ theme }) => css`
    text-align: center;
    color: ${theme.colors.secondary};
    font-size: ${theme.font.sizes.medium};
  `}
`;
