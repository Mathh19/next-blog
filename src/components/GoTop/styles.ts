import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`
    position: fixed;
    background-color: #082032;
    color: ${theme.colors.secondary};
    display: flex;
    align-items: center;
    justify-content: center;
    width: 4rem;
    height: 4rem;
    bottom: 2rem;
    right: 2rem;
    z-index: 6;
    opacity: 0.7;
    border-radius: 50%;
    border: 0.2rem solid ${theme.colors.secondary};
    &:hover {
      transition: 500ms ease-in-out all;
      background-color: #12324a;
      opacity: unset;
    }
  `}
`;
