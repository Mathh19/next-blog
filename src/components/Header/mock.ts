import { HeaderProps } from '.';

export default {
  srcImg: 'assets/images/cat-test.png',
  title: 'Titulo blog',
  description:
    'Ea aliqua labore anim minim voluptate est in cillum qui consequat cupidatat adipisicing ex in. Amet velit consequat mollit deserunt consectetur dolor exercitation commodo consequat voluptate eu non commodo ipsum.',
  showDescription: true,
} as HeaderProps;
