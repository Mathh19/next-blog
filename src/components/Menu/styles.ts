import styled, { css, DefaultTheme } from 'styled-components';
import { Title as HeadingStyles } from '../Heading/styles';

type MenuBehaviourProps = {
  menuVisible: boolean;
};

const wrapperChanger = (menuVisible: MenuBehaviourProps['menuVisible']) => css`
  left: ${menuVisible ? '0' : '-36rem'};
  overflow-y: ${menuVisible ? 'auto' : 'hidden'};
`;

export const Wrapper = styled.div<MenuBehaviourProps>`
  ${({ theme, menuVisible }) => css`
    background: ${theme.colors.bgMenu};
    padding: ${theme.spacings.large};
    display: flex;
    position: fixed;
    z-index: 1;
    width: 100%;
    max-width: 32rem;
    height: 100vh;
    top: 0;
    left: 0rem;
    transition: all 400ms ease-in-out;
    overflow-y: auto;
    box-shadow: 5px 0 10px -2px ${theme.colors.black};
    scrollbar-color: ${theme.colors.secondary} ${theme.colors.primary};
    scrollbar-width: thin;
    ${wrapperChanger(menuVisible)}


    ::-webkit-scrollbar {
      width: 10px;
    }

    ::-webkit-scrollbar-track {
      background: ${theme.colors.primary};
    }

    ::-webkit-scrollbar-thumb {
      background: ${theme.colors.secondary};
      border-radius: 10px;
    }

    ::-webkit-scrollbar-thumb:hover {
      background: #B21031;
    }
  `}
`;

export const Nav = styled.nav`
    margin: auto;
    width: 100%;
`;

export const Logo = styled.div`
  ${({ theme }) => css`
    ${HeadingStyles} {
      display: flex;
      justify-content: center;
      margin: 0;
      margin-bottom: ${theme.spacings.xxlarge};
    }

    img {
      border: 0.5rem solid ${theme.colors.secondary};
    }
  `}
`;

const buttonChanger = (
  menuVisible: MenuBehaviourProps['menuVisible'],
  theme: DefaultTheme,
) => css`
  left: ${menuVisible ? '26rem' : '1rem'};
  color: ${theme.colors.secondary};

  @media ((max-width: 300px)) {
    left: ${menuVisible ? '23rem' : '1rem'};
  }
`;

export const OpenClose = styled.a<MenuBehaviourProps>`
  ${({ theme, menuVisible }) => css`
    position: fixed;
    top: ${theme.spacings.medium};
    color: ${theme.colors.white};
    background: #082032;
    z-index: 2;
    width: 3rem;
    height: 3rem;
    left: 26rem;
    border-radius: 4px;
    transition: all 400ms ease-in-out;
    ${buttonChanger(menuVisible, theme)}
  `}
`;
