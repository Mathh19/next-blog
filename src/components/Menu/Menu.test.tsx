import { renderTheme } from '../../styles/render-theme';
import { Menu, MenuProps } from '.';

import mock from './mock';
import { fireEvent, screen } from '@testing-library/react';

const props: MenuProps = mock;

describe('<Menu />', () => {
  it('should render button link', () => {
    renderTheme(<Menu {...props} links={undefined} />);

    const buttonLink = screen.getByRole('link', { name: 'Open or Close menu' });
    const openMenuIcon = screen.getByLabelText('Open menu');

    expect(buttonLink).toBeInTheDocument();
    expect(openMenuIcon).toBeInTheDocument();

    expect(screen.queryByLabelText('Close menu')).not.toBeInTheDocument();
    expect(screen.queryByRole('navigation')).not.toBeInTheDocument();
  });

  it('should open/close menu button click', () => {
    renderTheme(<Menu {...props} />);

    const buttonLink = screen.getByRole('link', { name: 'Open or Close menu' });
    fireEvent.click(buttonLink);

    expect(screen.getByLabelText('Close menu')).toBeInTheDocument();
    expect(screen.queryByLabelText('Open menu')).not.toBeInTheDocument();
    expect(screen.getByRole('navigation')).toBeInTheDocument();
    expect(screen.getByRole('heading', { name: 'Matheus Freitas' }));
    expect(screen.getByRole('img', { name: 'Matheus Freitas' }));
    expect(
      screen.getByRole('navigation').querySelectorAll('a:not([href="/"])'),
    ).toHaveLength(mock.links.length);

    fireEvent.click(buttonLink);

    expect(screen.queryByLabelText('Close menu')).not.toBeInTheDocument();
    expect(screen.queryByRole('navigation')).not.toBeInTheDocument();
    expect(screen.getByLabelText('Open menu')).toBeInTheDocument();
  });

  it('should match snapshot', () => {
    const { container } = renderTheme(<Menu {...props} />);
    expect(container).toMatchSnapshot();
  });
});
