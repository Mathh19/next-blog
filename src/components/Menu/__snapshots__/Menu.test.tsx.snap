// Vitest Snapshot v1

exports[`<Menu /> > should match snapshot 1`] = `
.c6 {
  color: #082032;
  line-height: 1.2;
  font-size: 2.4rem;
  text-transform: uppercase;
}

.c2 {
  background: #082032;
  padding: 3.2rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: fixed;
  z-index: 1;
  width: 100%;
  max-width: 32rem;
  height: 100vh;
  top: 0;
  left: 0rem;
  -webkit-transition: all 400ms ease-in-out;
  transition: all 400ms ease-in-out;
  overflow-y: auto;
  box-shadow: 5px 0 10px -2px #000000;
  left: -36rem;
  overflow-y: hidden;
}

.c2::-webkit-scrollbar {
  width: 10px;
}

.c2::-webkit-scrollbar-track {
  background: #082032;
}

.c2::-webkit-scrollbar-thumb {
  background: #E94560;
  border-radius: 10px;
}

.c2::-webkit-scrollbar-thumb:hover {
  background: #720d1e;
}

.c3 {
  margin: auto;
  width: 100%;
}

.c4 .c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin: 0;
  margin-bottom: 4.8rem;
}

.c4 img {
  border: 0.5rem solid #E94560;
}

.c0 {
  position: fixed;
  top: 2.4rem;
  color: #FFFFFF;
  background: #082032;
  z-index: 2;
  width: 3rem;
  height: 3rem;
  left: 26rem;
  border-radius: 4px;
  -webkit-transition: all 400ms ease-in-out;
  transition: all 400ms ease-in-out;
  left: 1rem;
  color: #FFFFFF;
}

.c1 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
}

.c7 > img {
  width: 150px;
  height: 150px;
  border-radius: 50%;
}

.c8 {
  display: block;
  color: #FFFFFF;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin-bottom: 1.6rem;
  font-size: 1.8rem;
  -webkit-transition: all ease-in-out 100ms;
  transition: all ease-in-out 100ms;
}

.c8:hover {
  border-right: 0.5rem solid #E94560;
  color: #E94560;
}

@media ((max-width:300px)) {
  .c0 {
    left: 1rem;
  }
}

<div>
  <a
    aria-label="Open or Close menu"
    class="c0"
    href="#"
    title="Open or Close menu"
  >
    <svg
      aria-hidden="true"
      aria-label="Open menu"
      class="c1"
      fill="currentColor"
      focusable="false"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0V0z"
        fill="none"
      />
      <path
        d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
      />
    </svg>
  </a>
  <div
    aria-hidden="true"
    class="c2"
  >
    <nav
      class="c3"
    >
      <div
        class="c4"
      >
        <h1
          class="c5 c6"
        >
          <a
            class="c7"
            href="/"
            target="_self"
          >
            <img
              alt="Matheus Freitas"
              src="/assets/images/cat-test.png"
            />
          </a>
        </h1>
      </div>
      <a
        class="c8"
        href="/home/1"
        target="_self"
      >
        My jobs
      </a>
      <a
        class="c8"
        href="/home/2"
        target="_self"
      >
        About me
      </a>
      <a
        class="c8"
        href="/home/3"
        target="_self"
      >
        Services
      </a>
      <a
        class="c8"
        href="http://localhost/home/4"
        target="_self"
      >
        External Link
      </a>
      <a
        class="c8"
        href="/home/5"
        target="_self"
      >
        JavaScript
      </a>
      <a
        class="c8"
        href="/home/6"
        target="_self"
      >
        C#
      </a>
      <a
        class="c8"
        href="/home/7"
        target="_self"
      >
        Consectetur
      </a>
      <a
        class="c8"
        href="/home/8"
        target="_self"
      >
        Text test
      </a>
      <a
        class="c8"
        href="/home/9"
        target="_self"
      >
        Pac-man
      </a>
      <a
        class="c8"
        href="/home/10"
        target="_self"
      >
        Bungou Stray Dogs
      </a>
      <a
        class="c8"
        href="/home/11"
        target="_self"
      >
        A menu
      </a>
    </nav>
  </div>
</div>
`;
